File: Continuous Subarray Sum/Continuous_Subarray_Sum.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        if k == 0:
            for i in range(len(nums) - 1):
                if nums[i] == 0 and nums[i + 1] == 0:
                    return True
            return False
        
        remainder_map = {0: -1}  
        cumulative_sum = 0
        
        for i in range(len(nums)):
            cumulative_sum += nums[i]
            remainder = cumulative_sum % k
            
            if remainder in remainder_map:
                if i - remainder_map[remainder] > 1:
                    return True
            else:
                remainder_map[remainder] = i
                
        return False

File: Continuous Subarray Sum/Continuous_Subarray_Sum.cpp
class Solution {
public:
    bool checkSubarraySum(std::vector<int>& nums, int k) {
        if (k == 0) {
            for (int i = 0; i < nums.size() - 1; ++i) {
                if (nums[i] == 0 && nums[i + 1] == 0) {
                    return true;
                }
            }
            return false;
        }

        std::unordered_map<int, int> remainderMap;
        remainderMap[0] = -1; 
        int accumulativeSum = 0;
        
        for (int i = 0; i < nums.size(); ++i) {
            accumulativeSum += nums[i];
            int remainder = accumulativeSum % k;

            if (remainder < 0) {
                remainder += k;
            }

            if (remainderMap.find(remainder) != remainderMap.end()) {
                if (i - remainderMap[remainder] > 1) {
                    return true;
                }
            } else {
                remainderMap[remainder] = i;
            }
        }
        
        return false;
    }
};
